src/list-names.py

```diff
@@ -1,4 +1,16 @@
 import argparse
+from typing import TextIO
+
+
+def make_library(f: TextIO) -> dict[str, str]:
+    out: dict[str, str] = {}
+    tmp = ''.join(f.readlines())
```

Comment:
If you just want the content of a file, `f.read()` is the method you want. Getting all the lines and concatenating them is a rather slow way of readine the file into a single string.